packages/block-editor/README.md

```diff
@@ -228,10 +228,11 @@ _Parameters_
 
 -   _props_ `Object`:
 -   _props.clientId_ `string`: Client ID of block.
+-   _props.maximumLength_ `number`: The maximum length that the block title string may be before truncated.
 
 _Returns_
 
--   `?string`: Block title.
+-   `JSX.Element`: Block title.
```

Comment:
Ah, good catch. I did a quick double-take on that one, because the _component_ is returning a `string`, but of course React renders it as a `JSX.Element`, so that's a more accurate description of what it's returning. Is that right?

Reply:
Yes, nice spotting. Actually this was an automated update. 

I left it there since the component is currently used as a component, so `<BlockTitle />`. It's used by the Block List Tree in this way for example.

But of course `BlockTitle( { clientId } )` will return a plain string.

I did hesitate before splitting the component into component + hook and just calling `BlockTitle( { clientId } )`, but perceived convention prevented me.

What do you reckon?

Reply:
Another option that occurred to me would be to refactor `<BlockTitle />` to be a regular function.

It _ostensibly_ appears to be a component, but only due to the way it's being used.

Given that it doesn't return any markup maybe it's a possible follow up. 🤷

Reply:
> Actually this was an automated update.

Ah, nice!

> What do you reckon?

I think I'd lean toward the approach you've gone with in this PR — abstract away the internals of that component so that you've got reusability for the case of the block settings dropdown, then we could look at refactoring `BlockTitle` separately if need be. I suppose there's nice ergonomics of adding it to a component via `<BlockTitle clientId={ clientId } />` rather than having to call a function within JSX like `{ getBlockTitle( clientId ) }`?. Also, it looks like the `BlockTitle` component is used in half a dozen places, so it might be better to leave it as a component (at least for the time being) 🙂

Reply:
Great, thanks for your thoughts. Sounds very reasonable 👍